testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/group/GroupTest.java

```diff
@@ -1047,7 +1047,75 @@ public void getGroupsWithFullRepresentation() {
         assertFalse(groups.isEmpty());
         assertTrue(groups.get(0).getAttributes().containsKey("attribute1"));
     }
-    
+
+
+    @Test
+    public void searchGroupsByNameContaining() {
+        RealmResource realm = adminClient.realms().realm("test");
+        GroupsResource groupsResource = adminClient.realms().realm("test").groups();
+
+        GroupRepresentation group1 = new GroupRepresentation();
+        group1.setName("group-name-1");
+
+        GroupRepresentation group2 = new GroupRepresentation();
+        group2.setName("group-name-2");
+
+
+        Map<String, List<String>> attributes = new HashMap<String, List<String>>();
+        group1 = createGroup(realm, group1);
+        group2 = createGroup(realm, group2);
+
+        List<GroupRepresentation> groups = groupsResource.groups("group-name", false, 0, 20, false);
+
+        assertFalse(groups.isEmpty());
+        assertEquals(2,groups.size());
```

Comment:
Could you leverage `Creator.create(RealmResource, GroupRepresentation)` instead in the newly added tests to clean up after the test completes?

Example:
https://github.com/keycloak/keycloak/blob/fd5a423b551b9836bda44d47b38d7cd34f8d27c4/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/UserTest.java#L2883-L2894

Also prefer `assertThat` to `assertFalse` / `assertEquals` - upon failure, the output providers information about the details of failed assertion, and it is also shorter:

```java
        assertThat(groups, Matchers.hasSize(2));
```

Reply:
I have no problem with moving over to assertThat, though I do want to remark that it is deprecated before doing so

Reply:
`assertThat` is still the preferred way for assertions.
`org.hamcrest.MatcherAssert.assertThat` supersedes `org.junit.Assert.assertThat`. Feel free to keep it as is or fix the import along the way.

Reply:
I will update my usages and fix the imports, removing the star import for junit in the process

Reply:
updated usages